internal/driver/driver.go

```diff
@@ -727,10 +703,33 @@ func (d *Driver) RefreshAllDevicePaths() {
 	}
 }
 
+// Deprecated: 3.0 version of the service supports Path which is a string instead of Paths which is []string which leads to breaking change.
+// updatePathToPaths takes care of this breaking change.
+func (d *Driver) updatePathToPaths(cd models.Device) {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	if value, ok := cd.Protocols[UsbProtocol][Path]; ok {
+		if value != nil {
+			cd.Protocols[UsbProtocol][Paths] = []string{value.(string)}
+			delete(cd.Protocols[UsbProtocol], "Path")
+
+			if err := d.ds.PatchDevice(dtos.UpdateDevice{
+				Name:      &cd.Name,
+				Protocols: dtos.FromProtocolModelsToDTOs(cd.Protocols),
+			}); err != nil {
+				d.lc.Errorf("failed to update the device %s, error: %s", cd.Name, err.Error())
+			}
+		}
+	}
+}
+
 // RefreshDevicePaths checks whether the existing device matches the connected device.
 // If there is a mismatch between them, scan all paths to find the matching device
 // and update the existing device with the correct path.
 func (d *Driver) RefreshDevicePaths(cd models.Device) {
+	d.updatePathToPaths(cd)
```

Comment:
Is this the only place that needs to call this?

Reply:
yes this function is called when the service starts up and also when the discovery is called.